articles/user-guide/using-qsharp/control-flow.md

```diff
@@ -0,0 +1,329 @@
+This may be followed by any number of else-if clauses, each of which consists of the keyword `elif`, an open parenthesis `(`, a Boolean expression, a close parenthesis `)`, and a statement block (the _else-if_ block).
+Finally, the statement may optionally finish with an else clause, which consists of the keyword `else` followed by another statement block (the _else_ block).
+
+The `if` condition is evaluated, and if it is true, the then block is executed.
+If the condition is false, then the first else-if condition is evaluated; if it is true, that else-if block is executed.
+Otherwise, the second else-if block is tested, and then the third, and so on until either a clause with a true condition is encountered or there are no more else-if clauses.
+If the original if condition and all else-if clauses evaluate to false, the else block is executed if one was provided.
+
+Note that whichever block is executed is executed in its own scope.
+Bindings made inside of an `if`, `elif`, or `else` block are not visible after its end.
+
+For example,
+
+```qsharp
+if (result == One) {
+    X(target);
+} 
+```
+or
+```qsharp
+if (i == 1) {
+    X(target);
+} elif (i == 2) {
+    Y(target);
+} else {
+    Z(target);
+}
+```
+
+## For Loop
+
+The `for` statement supports iteration over an integer range or over an array.
+The statement consists of the keyword `for`, an open parenthesis `(`, followed by a symbol or symbol tuple, the keyword `in`, an expression of type `Range` or array, a close parenthesis `)`, and a statement block.
+
+The statement block (the body of the loop) is executed repeatedly, with the defined symbol(s) (the loop variable(s)) bound to each value in the range or array.
+Note that if the range expression evaluates to an empty range or array, the body will not be executed at all.
+The expression is fully evaluated before entering the loop, and will not change while the loop is executing.
+
+The loop variable is bound at each entrance to the loop body, and unbound at the end of the body.
+In particular, the loop variable is not bound after the for loop is completed.
+The binding of the declared symbol(s) is immutable and follows the same rules as other variable bindings. 
+
+For some examples, supposing `qubits` is a register of qubits (i.e. of type `Qubit[]`), 
+
+```qsharp
+// ...
+for (qubit in qubits) {  // iterate over each qubit value in the array qubits
+    H(qubit);
+}
+
+mutable results = new (Int, Results)[Length(qubits)];
+for (index in 0 .. Length(qubits) - 1) {  // iterates over the integers in the Range 0 .. (Length(qubits) - 1)
+    set results w/= index <- (index, M(qubits[index])); // measure each qubit, updates the tuple value in the results array that at index 
+}
+
+mutable accumulated = 0;
+for ((index, measured) in results) { // iterates over the tuple values in results
+    if (measured == One) {
+        set accumulated += 1 <<< index;
+    }
+}
+```
```

Comment:
I agree with the readability issue, but I wasn't sure about doing it here because they all actually build on one another (i.e. using objects created in the previous loop)